File: dcp-solutions/2025-02-12/solution.php
```php
function majorityElement($lst) {
    $count = array();
    foreach ($lst as $num) {
        if (!isset($count[$num])) {
            $count[$num] = 0;
        }
        $count[$num]++;
    }

    foreach ($count as $num => $occurs) {
        if ($occurs > floor(count($lst) / 2)) {
            return $num;
        }
    }
}
```

File: dcp-solutions/2025-02-12/solution.rb
```ruby
def majority_element(lst)
  count = Hash.new(0)
  lst.each do |num|
    count[num] += 1
  end
  list, threshold = [], (lst.length.to_f / 2.0).ceil
  lst.each do |num|
    if count[num] >= threshold
      return num
    end
  end
end
```